patches/tModLoader/Terraria/ModLoader/Default/UnloadedInfo.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Terraria.ModLoader.IO;
+
+namespace Terraria.ModLoader.Default
+{
+	internal class UnloadedInfo
+	{
+		public string modName;
+		public string name;
+		internal ushort FallbackType;
```

Comment:
Make these readonly properties and use UpperCamelCase.

Reply:
Not sure about this one. 
Readonly wouldn't work with current subclasses for the different tile types constructing base values afaik (namely looking at chests while thinking this). I've swapped from public to internal, but that's not getting to what you're looking for I presume.

Per Mirsario's comment, these are attributes so camelCase not Upper.

Reply:
I was mainly talking about the public ones fields, the conventions for attributes is to use public properties rather than public fields.
As for `internal`, everywhere I've seen them used, they follow nearly the same conventions as `public` because they are practically the same except you have to whitelist which external assemblies can use them. In this case, these attributes are meant to be used outside of the class itself so a property would just make more sense then a field and be cleaner.

As for readonly, I understand what you mean about not being able to be readonly. I wasn't able to see much else that so I was just putting it out there as something to check.

Does make me wonder if there would be a better way to handle it from an object oriented approach to make it more protected. Setting a base field in an extending class outside of its constructor is generally a suboptimal approach.

Reply:
Thanks for the education there Rartrin; I didn't quite understand the difference between them all as I've only started to do OOP this year. Makes a lot more sense now; and I'm all for safe coding practices.

I made them readonly when I refactored the UnloadedInfo back down to one class, so that's been fixed. Realized I could just store a customData tagcompound in Info for chests and could reduce the classes from there.

Thanks for the review!